org.jacoco.core.test/src/org/jacoco/core/test/filter/targets/Constructor.java

```diff
@@ -36,7 +36,8 @@ private Inner() { // $line-inner$
 	}
 
 	private static class InnerStatic {
-		private Object field = this;
+		@SuppressWarnings("unused")
+		private final Object field = this;
```

Comment:
I did cherry-pick of this change separately into master as it is unrelated to `MethodAnalyzerTest` and `MethodAnalyzer`.